
<launch>
	<!-- <arg name="leader_robot_name" value="robot_0"/>
	<arg name="slave_robot" value=""/>

	<arg name="slave_x" value="-1.5"/> 
	<arg name="slave_y" value="1.5"/>  -->

	<node pkg="tf" type="static_transform_publisher" name="base_to_$(arg slave_robot)" args="$(arg slave_x) $(arg slave_y) 0 0 0 0   $(arg leader_robot_name)/base_footprint $(arg leader_robot_name)/$(arg slave_robot) 100" />
    
    <param name="max_vel_x" value="0.8" />
    <param name="min_vel_x" value="0.05" />
    <param name="max_vel_theta" value="0.8" />
    <param name="min_vel_theta" value="0.05" />
    
    <param name = "tf_prefix" value="$(arg slave_robot)"/>
	
    <node pkg="turtlebot_sim" type="leader_follower_formation_pid.py" name="leader_follower_formation_pid" output="screen" >
        <param name="leader_robot_name" value="$(arg leader_robot_name)" />
        <param name="follower_robot_name" value="$(arg slave_robot)" />
        <param name="k_v" type="double" value="1.5" />
        <param name="k_l" type="double" value="1.0" />
        <param name="k_a" type="double" value="1.0" />
        <param name="slave_x" type="double" value="$(arg slave_x)" />
        <param name="slave_y" type="double" value="$(arg slave_y)" />
        <param name="base_to_slave" type="string" value="/$(arg slave_robot)" /> <!-- slave target的topic名稱 -->
        <param name="base_frame" type="string" value="base_footprint" />
    </node> 
</launch>
